feature_engine/discretisation/decision_tree.py

```diff
@@ -69,13 +72,13 @@ class DecisionTreeDiscretiser(BaseNumericalTransformer):
 
     def __init__(
         self,
-        cv=3,
-        scoring="neg_mean_squared_error",
-        variables=None,
-        param_grid=None,
-        regression=True,
-        random_state=None,
-    ):
+        cv: int = 3,
+        scoring: str = "neg_mean_squared_error",
+        variables: Optional[List[str]] = None,
+        param_grid: Optional[dict] = None,
```

Comment:
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html
# For mappings, we need the types of both keys and values
x: Dict[str, float] = {'field': 2.0}

To determine the values, we need to look at the params that the tree regressor or classifier can take. I don't remember of the top of my head if all are numerical, or there are as well some boolean, and strings

Reply:
update type annotations